autoeq/fixed_band.go

```diff
@@ -6,34 +6,50 @@ import (
 	"strings"
 )
 
-const preampFields = 12
+const fixedBandFields = 12
 
-// FixedBandEQ represents a single EQ band.
-type FixedBandEQ struct {
+// FixedBandFilter represents a single EQ band.
+type FixedBandFilter struct {
 	Frequency int     // Hz
-	Gain      float64 // Db
+	Gain      float64 // dB
 	Q         float64 // fixed
 }
 
-// FixedBandEQs represents a simple fixed bands EQ.
-type FixedBandEQs []*FixedBandEQ
+// FixedBandEQ represents a simple fixed bands EQ.
+type FixedBandEQ struct {
+	Filters []*FixedBandFilter
+	Preamp  float64
+}
 
 // ToFixedBandEQs transforms the raw EQ data into a fixed band EQ.
 // Returns an error if any.
-func ToFixedBandEQs(data []byte) (FixedBandEQs, error) {
+func ToFixedBandEQs(data []byte) (*FixedBandEQ, error) {
 	rows := strings.Split(string(data), "\n")
-	eqs := make(FixedBandEQs, len(rows))
+
+	fbEQ := &FixedBandEQ{
+		Filters: make([]*FixedBandFilter, len(rows)),
+	}
+
+	startIdx := 0 // rows index, increment if first row is preamp
+
+	// parse preamp
+	if strings.HasPrefix(rows[0], "Preamp") {
```

Comment:
bound checks missing

Reply:
line below as well

Reply:
> bound checks missing

I didn't add a bound check there because `rows` will always have length of 1 if the separator is not empty, because of how [strings.Split](https://pkg.go.dev/strings#Split) works.
You can have a look at it on [Go playground](https://go.dev/play/p/j7m3vaPLHJq).

Will add a check on the result of `strings.Fields` 😉